builtin/mainmenu/dlg_settings_advanced.lua

```diff
@@ -774,17 +774,38 @@ local function create_change_setting_formspec(dialogdata)
 	)
 end
 
+local function set_setting_if_changed(name, type, new_value)
+	if type == "bool" and new_value ~= core.settings:get_bool(name) then
```

Comment:
```suggestion
	if type == "boolean" and new_value ~= core.settings:get_bool(name) then
```
See http://www.lua.org/manual/5.1/manual.html#pdf-type.

Reply:
`type` isn't a Lua type. It's an attribute stored in the `settings` table (declared in file-local scope), which in turn is populated from `settingtypes.txt`. See also, https://github.com/minetest/minetest/blob/dea5a152b2c80276a0f09197e06e411a2ec53374/builtin/mainmenu/dlg_settings_advanced.lua#L780

Reply:
Oh, ok, sorry.